exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/union/UnionAllRecordBatch.java

```diff
@@ -162,6 +162,25 @@ private IterOutcome doWork() throws ClassTransformationException, IOException, S
     allocationVectors = Lists.newArrayList();
     transfers.clear();
 
+    // If both sides of Union-All are empty
+    if(unionAllInput.isBothSideEmpty()) {
+      for(int i = 0; i < outputFields.size(); ++i) {
```

Comment:
Ok, I suppose we are not supporting the SELECT * case, so the output fields only depends on what is specified in the SELECT list, not from the table.